lib/db/schema/patch-019-020.sql

```diff
@@ -0,0 +1,95 @@
+-- This migration:
+--
+-- * Adds a column, devices.callbackPublicKey
+-- * Updates the accountDevices stored procedure with the new column
+-- * Adds a unique constraint on devices.uid, devices.sessionTokenId
+-- * Removes the (never-used) upsertDevice stored procedure
+-- * Adds two stored procedures, createDevice and updateDevice
+
+ALTER TABLE devices
+ADD COLUMN callbackPublicKey BINARY(32),
+ADD CONSTRAINT UQ_devices_sessionTokenId UNIQUE (uid, sessionTokenId);
+
+CREATE PROCEDURE `accountDevices_3` (
+  IN `inUid` BINARY(16)
+)
+BEGIN
+  SELECT
+    d.uid,
+    d.id,
+    d.sessionTokenId,
+    d.name,
+    d.type,
+    d.createdAt,
+    d.callbackURL,
+    d.callbackPublicKey,
+    s.uaBrowser,
+    s.uaBrowserVersion,
+    s.uaOS,
+    s.uaOSVersion,
+    s.uaDeviceType,
+    s.lastAccessTime
+  FROM
+    devices d LEFT JOIN sessionTokens s
+  ON
+    d.sessionTokenId = s.tokenId
+  WHERE
+    d.uid = inUid;
+END;
+
+DROP PROCEDURE `upsertDevice_1`;
```

Comment:
Should be safe, but but it might make e.g. rollbacks harder.  I suggest just leaving it in place and we'll garbage-collect alongside other obsolete stored procedures.